src/pins/gen.rs

```diff
@@ -97,13 +106,26 @@ macro_rules! pins {
             }
         }
 
-        /// A token representing a pin
+        /// A token representing a pin.
+        /// Tokens are unique for each pin, and ensure that no pins are re-used by accident.
+        /// They also hold the `pin::Trait` information for their associated pin.
         ///
         /// Used by [`GPIO`] to uphold correctness guarantees. Please refer to
         /// [`GPIO`] for more information.
         ///
         /// [`GPIO`]: ../gpio/struct.GPIO.html
-        pub struct Token<T, State>(T, PhantomData<State>);
+        pub struct Token<Trait: 'static, State>(&'static Trait, PhantomData<State>);
+
+        impl<Trait, State> Token<Trait, State>
+        {
+            /// Get identifying information about the pin representred by this token
+            /// in the form of a [`pins::Trait`]
+            ///
+            /// [`pins::Trait`]: trait.Trait.html
+            pub fn pin_info(&self) -> &Trait{
+                self.0
+            }
+        }
```

Comment:
Since we now throw away the token when initializing a GPIO pin, I don't think we need any of this any more.

```suggestion
        pub struct Token<Pin, State>(PhantomData<Pin>, PhantomData<State>);
```

Reply:
🤯